### PR TITLE
[13.0][MIG] hr_expense_payment_difference

### PR DESCRIPTION
**Pull request by ps-tubtim**
Standard migration and move `hr_expense_sheet_register_payment.py` from _model_ to _wizard_ folder.

### CODE REVIEW

**Comment by kittiu** [comment]
Tested not yet ok. @ps-tubtim can you check?

Here is the step.

1. Just a simple payment, at $100 and deduct $1
image

2. State is not Paid.
image

3. Only reconcile AP $99
image

Note: if I pay full 100, it is "Paid".

File: hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            [("type", "=", "bank")], limit=1
+        )
+        cls.account_id = cls.env["account.account"].search([], limit=1)
+
+        employee_home = cls.env["res.partner"].create({"name": "Employee Home Address"})
+        cls.employee_1 = cls.env["hr.employee"].create(
+            {"name": "Employee 1", "address_home_id": employee_home.id}
+        )
+        cls.expense_1 = cls.create_expense(cls, "Expense Test1")
+        cls.expense_2 = cls.create_expense(cls, "Expense Test2")
```
Comment (kittiu) [inline hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py]: If not used, please remove.

File: hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+        PaymentWizard = self.env["hr.expense.sheet.register.payment.wizard"]
+        with Form(PaymentWizard.with_context(ctx)) as f:
+            f.journal_id = self.journal_bank
+            f.amount = self.product.standard_price - 20.0
+            f.payment_difference_handling = "reconcile"
+            f.writeoff_account_id = self.account_id
+        payment_wizard = f.save()
+        self.assertEqual(payment_wizard.payment_difference, 20.0)
+        payment_wizard.expense_post_payment()
```
Comment (kittiu) [inline hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py:65]: Test to make sure it is "Paid"
```suggestion
        payment_wizard.expense_post_payment()
        self.assertEqual(self.sheet.state, "done")
```

**Comment by kittiu** [comment]
Please also add enhancement from
https://github.com/OCA/hr/pull/909

File: hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py (hunk 1 above)
Comment (ps-tubtim) [reply hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py]: Removed.

File: hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py (hunk 2 above)
Comment (ps-tubtim) [reply hr_expense_payment_difference/tests/test_hr_expense_payment_difference.py:65]: Added.

**Comment by ps-tubtim** [comment]
@kittiu Thank you for your review. All reviews are done. Could you review it again?

**Comment by kittiu** [comment]
Functional & Code reivew, LGTM :+1:

**Review by kittiu — APPROVED** [review]

**Review by Saran440 — APPROVED** [review]
Functional Test :+1:

**Pull request merged**